src/Bufferer/PipeBufferer.php

```diff
@@ -76,6 +88,15 @@ public function isBuffering(): bool
         return $this->buffering;
     }
 
+    public function waitForWrite(): Promise
+    {
+        if (!$this->deferred) {
+            $this->deferred = new Deferred;
```

Comment:
Only if the client is consuming bytes faster than your providing them, so not every write-respond cycle. This is a very minor price to be paid, I doubt it has any noticeable affect on performance.

Reply:
> Only if the client is consuming bytes faster than your providing them

This will be very common in practice though. But yeah probably impact isn't large.